pep-9999.txt

```diff
@@ -0,0 +1,92 @@
+PEP: 9999
+Title: Metadata for Python Software Packages 1.3
+Version: $Revision$
+Last-Modified: $Date$
+Author: Dustin Ingram <di@di.codes>
+Discussions-To: Distutils SIG
+Status: Accepted
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 1-Dec-2017
+Python-Version: 3.x
+Post-History:
+
+
+Abstract
+========
+
+This PEP describes a mechanism for adding metadata to Python distributions.
+It includes specifics of the field names, and their semantics and
+usage.
+
+This document specifies version 1.3 of the metadata format.
+Version 1.0 is specified in PEP 241.
+Version 1.1 is specified in PEP 314.
+Version 1.2 is specified in PEP 345.
+
+Version 1.3 of the metadata format changes the canonical source for field
+specification to the `Core Metadata Specification`_.
+
+Fields
+======
+
+The canonical source for the names and semantics of each of the supported
+metadata fields is the `Core Metadata Specification`_ document.
+
+Fields marked with "(Multiple use)" may be specified multiple times in a single
+PKG-INFO file.  Other fields may only occur once in a PKG-INFO file.  Fields
+marked with "(optional)" are not required to appear in a valid PKG-INFO file;
+all other fields must be present.
+
+Version Specifiers
+==================
+
+Version numbering requirements and the semantics for specifying comparisons
+between versions are defined in PEP 440.
+
+Environment markers
+===================
+
+An **environment marker** is a marker that can be added at the end of a
+field after a semi-colon (";"), to add a condition about the execution
+environment.
+
+The environment marker format used to declare such a condition is defined in
+the environment markers section of PEP 508.
+
+Summary of Differences From PEP 345
+===================================
+
+* Metadata-Version is now 1.3.
+
+* Fields are now specified via the `Core Metadata Specification`_.
```

Comment:
Additional changes to cover:

- `Name` format is defined in https://www.python.org/dev/peps/pep-0508/#names
- new field `Description-Content-Type`
- new field `Provides-Extra`

Reply:
Could you explain why you think this PEP should explicitly include the two new fields? I opted not to because I felt that it somewhat defeats the purpose of this PEP (i.e. that new fields are now specified elsewhere, not in this PEP.)

Do you think this PEP should include the entire specification for the two new fields, or just mention here (in the "Summary of Differences" section) that they have been added? 

In your mind, if we needed to add another field after the acceptance of this PEP, what would happen? Would we amend this PEP? Or would that field be part of "Metadata 1.4"? Would a new metadata version require a new PEP?

Along those lines, should this PEP mention something about when and why the metadata version number should change?

Reply:
The goal of the process change is to switch to a model where distutils-sig PEPs describe *changes* to the specifications, but are not themselves used as primary reference documents. The big difference that makes is that it means we don't copy and paste *unchanged* fields into the new PEP - we only list the new & updated ones.

Since this is a backfill PEP, you're creating a record of past decisions already made in GitHub PRs rather than actually requesting approval of the changes, but in any future metadata update PEPs will also need to make the argument for why particular additions or changes are needed.

For moving the reference URLs to PyPUG, the appropriate reference would be: https://github.com/pypa/pypa.io/issues/11

For Description-Content-Type the appropriate reference would be: https://github.com/pypa/python-packaging-user-guide/pull/258

For Provides-Extra, the most appropriate reference would be probably be to an early draft of PEP 426, such as: https://github.com/python/peps/blob/493ff3b7b5e6f83e7ebaf6ac8902fcdc8835abbc/pep-0426.txt#L428

Reply:
A shorter version of that comment: rather than being about what the field *is*, the entry in the PEP should focus on *why* we added it.

Provides-Extra: so installers know which extras are available for selection, and so publishing tools can check for typos in environment markers.
Description-Content-Type: so tools (most notably PyPI) can be told how to render the long description, rather than always assuming the use of reStructuredText